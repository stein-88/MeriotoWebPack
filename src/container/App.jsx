import React, { useState } from 'react'
import {
 BrowserRouter, Routes, Route, Navigate,
} from 'react-router-dom'
import Homepage from '@components/Homepage'
import RegProduct from '@components/RegProduct'
import Show from '@components/Show'
import NewShow from '@components/NewShow'
import Modal from '@generics/Modal'
import Footer from '@components/Footer'

const App = () => {
    const [content, setContent] = useState(undefined)

    const handleContent = (param) => setContent(param)

    return (
        <BrowserRouter>
            <Routes>
                <Route element={<Homepage />} path="/" />
                <Route element={<RegProduct handleContent={handleContent} />} path="/registerproduct" />
                <Route element={<RegProduct isEdit handleContent={handleContent} />} path="/editproduct" />
                <Route element={<Show handleContent={handleContent} />} path="/showproduct" />
                <Route element={<NewShow />} path="/derickapi" />
                <Route element={<Navigate replace to="/" />} path="*" />
            </Routes>
            <Footer />
            <Modal content={content} />
        </BrowserRouter>
    )
}

export default App
