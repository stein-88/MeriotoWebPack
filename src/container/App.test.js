import React from 'react'
import renderer from 'react-test-renderer'
import App from './App'
import schemawaste from '../components/HomePage/ReduceWaste/data'
import schemaGenerate from '../components/HomePage/Generate/data'
const { text } = schemaGenerate
const { card } = schemawaste

describe('testing App', () => {
	const compHomePage = renderer.create(<App />)
	test('Should render without props', () => {
		const jsonHomePage = compHomePage.toJSON()
		expect(jsonHomePage).toMatchSnapshot()
	})
	
    test('Sould have a text', () => {
        const Result = compHomePage.root.findAllByProps({ className: "py-5 text-center text-white" })
        expect(Result).toHaveLength(1)
    })
	test('Should have 1 title and 2 cards', () => {
		const resultTitle = compHomePage.root.findAllByProps({ className: 'h1 pt-5' })
		const resultTxt = compHomePage.root.findAllByProps({ className: 'h3 my-4' })
		expect(resultTxt).toHaveLength(2)
		expect(resultTitle).toHaveLength(1)
	})
	test('Should have 1 img, 1 title and 2 texts', () => {
		const resultTxt = compHomePage.root.findAllByProps({ className: 'h3 mb-4' })
		const resultTitle = compHomePage.root.findAllByProps({ className: 'h1 py-5 mb-0' })
		const resultimg = compHomePage.root.findAllByProps({ className: 'img-fluid w-100' })
		expect(resultTxt).toHaveLength(text.length)
		expect(resultTitle).toHaveLength(1)
		expect(resultimg).toHaveLength(1)
	})
	test('Should have 1 img, 1 title and 1 button', () => {
		const resultIMG = compHomePage.root.findAllByProps({ className: 'imgheader' })
		const resultTitle = compHomePage.root.findAllByProps({ className: 'py-4 h1' })
		const resultBtn = compHomePage.root.findAllByProps({ className: 'botaoreg' })
		expect(resultIMG).toHaveLength(1)
		expect(resultTitle).toHaveLength(1)
		expect(resultBtn).toHaveLength(1)
	})
	test('Should have 1 title and 2 cards', () => {
		const resultCARDS = compHomePage.root.findAllByProps({ className: 'textobaixo' })
		expect(resultCARDS).toHaveLength(card.length)
	})
	test('Should have 3 images and 2 title', () => {
		const resultIMG = compHomePage.root.findAllByProps({ className: 'img-fluid' })
		const resultTitle = compHomePage.root.findAllByProps({ className: 'h1 mt-5 py-4 text-muted' })
		expect(resultIMG).toHaveLength(3)
		expect(resultTitle).toHaveLength(2)
	})
})