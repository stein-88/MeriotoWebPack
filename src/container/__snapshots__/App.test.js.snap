// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App Should render without props 1`] = `
Array [
  <section
    className="w-100"
  >
    <header>
      <section
        className="bgblack w-100 py-4 mb-5"
      >
        <section
          className="imgheader"
        />
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-12 text-center text-white"
            >
              <h2
                className="py-4 h1"
              >
                Join the Reuse Revolution for Free items!
              </h2>
              <a
                className="botaoreg"
                href="https://clavo.globechain.com/register"
              >
                <strong>
                  Register
                </strong>
              </a>
            </div>
          </div>
        </div>
      </section>
    </header>
    <main>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-12 text-muted"
          >
            <h2
              className="h1 pt-5"
            >
              Globechain the ESG data reuse solution making waste a resource for everyone
            </h2>
            <p
              className="h3 my-4"
            >
              We are the leading ESG data reuse marketplace connecting enterprises with non profits, small businesses and people to redistribute unneeded items.
            </p>
            <p
              className="h3 my-4"
            >
              We generate ESG data on your impact, help you reduce waste costs and achieve your sustainability and SDG targets.
            </p>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-12 text-muted"
          >
            <h2
              className="h1 py-5 mb-0"
            >
              Generate Impact
            </h2>
            <div
              className="col-12"
            >
              <section
                className="w-100 blocopc"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-xl-8 col-lg-6 py-4 text-white"
                  >
                    <p
                      className="h3 mb-4"
                    >
                      Our marketplace allows you to effortlessly manage and track your items, reduce costs, increase transparency and generate ESG data for your reporting.
                    </p>
                    <p
                      className="h3 mb-4"
                    >
                      We offer external reuse via our marketplace, internal reuse and loaning asset inventory system.
                    </p>
                  </div>
                  <div
                    className="col-12 col-xl-4 col-lg-6"
                  >
                    <img
                      alt="pc"
                      className="img-fluid w-100"
                      src="test-file-stub"
                    />
                  </div>
                </div>
              </section>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-12"
          >
            <h2
              className="h1 mt-5 py-4 text-muted"
            >
              Get Reusing
            </h2>
            <section
              className="pt-4 text-center w-100"
            >
              <img
                alt="companies"
                className="img-fluid"
                src="test-file-stub"
              />
              <img
                alt="companies"
                className="img-fluid"
                src="test-file-stub"
              />
              <img
                alt="companies"
                className="img-fluid"
                src="test-file-stub"
              />
            </section>
          </div>
        </div>
        <section>
          <div
            className="row"
          >
            <div
              className="col-12"
            >
              <h2
                className="h1 mt-5 py-4 text-muted"
              >
                Reduce Waste, Generate ESG Impact
              </h2>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-lg-6 col-sm-12 col-md-12"
            >
              <section
                className="sectionobrigatoria"
              >
                <a
                  href="https://clavo.globechain.com/corporates"
                >
                  <section
                    className="position-relative seccard secforest"
                  >
                    <section
                      className="textobaixo"
                    >
                      <h3
                        className="h2 pl-4 pr-4 text-white"
                      >
                        <strong>
                          Discover the benefits of listing items
                        </strong>
                      </h3>
                      <section
                        className="text-right w-100"
                      >
                        <span
                          className="mr-4 stylespan"
                        >
                          Learn more
                        </span>
                      </section>
                    </section>
                  </section>
                </a>
              </section>
            </div>
            <div
              className="col-lg-6 col-sm-12 col-md-12"
            >
              <section
                className="sectionobrigatoria"
              >
                <a
                  href="https://clavo.globechain.com/charities"
                >
                  <section
                    className="position-relative seccard seckid"
                  >
                    <section
                      className="textobaixo"
                    >
                      <h3
                        className="h2 pl-4 pr-4 text-white"
                      >
                        <strong>
                          How taking items benefits everyone
                        </strong>
                      </h3>
                      <section
                        className="text-right w-100"
                      >
                        <span
                          className="mr-4 stylespan"
                        >
                          Learn more
                        </span>
                      </section>
                    </section>
                  </section>
                </a>
              </section>
            </div>
          </div>
        </section>
      </div>
    </main>
  </section>,
  <footer>
    <section
      className="py-5 bgblack"
      id="footer"
    >
      <p
        className="py-5 text-center text-white"
      >
        Â© 2022 Globechain. All Rights Reserved.
      </p>
    </section>
  </footer>,
]
`;
